### ISSUE TITLE
[no-unused-vars] Don’t mark `declare class` expressions as unused

### ISSUE THREAD

**Issue by j-f1** [typescript-eslint/typescript-eslint#106]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error", { "args": "none" }]
  }
}
```

```TS
declare class ResizeObserver {
  public constructor(cb: (entries: ReadonlyArray<IResizeObserverEntry>) => void)

  public disconnect(): void
  public observe(e: HTMLElement): void
}
```

**Expected Result**

No error

**Actual Result**

```
.../desktop/app/src/lib/globals.d.ts
  220:15  error  'ResizeObserver' is defined but never used  @typescript-eslint/no-unused-vars
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.0.0` |
| `@typescript-eslint/parser`        | `1.0.0` |
| `ESLint`                           | `5.8.0` |
| `node`                             | `8.12.0` |
| `npm`                              | ??? |

**Comment by armano2** [comment]
we have a lot of issues with declarations, i solved some of them in #88
but i'm starting to think that we should turn this rule off if we are in declaration file.

**Comment by armano2** [comment]
i think we should move this to parser as @mysticatea proposed in some old PR in https://github.com/eslint/typescript-eslint-parser/pull/558

with that we could control it better

**Comment by armano2** [comment]
i just realized that #88 is leading to some false positives
```
var baz = 2
declare namespace Foo {
  var foo: bar.baz
}
```

`baz` is going to be marked as used

**Comment by j-f1** [comment]
Not sure how moving the check to the parser would help as the rule is still ignoring `markVariableAsUsed`.

**Comment by armano2** [comment]
i was thinking a little to far ahead, and didn't explain it (i tend to do so)

`ResizeObserver` is defined in 2 scopes `class` and `module`

`markVariableAsUsed` is setting eslintUsed for `class` scope

scope of variable is determined by https://github.com/eslint/eslint/blob/master/lib/linter.js#L522
`const currentScope = getScope(scopeManager, currentNode);`
